WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/CardPresentPaymentsOnboardingViewModel.swift

```diff
@@ -143,6 +143,14 @@ final class CardPresentPaymentsOnboardingViewModel: ObservableObject, PaymentSet
     var didUpdate: (() -> Void)? = nil
 }
 
+extension CardPresentPaymentsOnboardingViewModel: Equatable {
+    static func == (lhs: CardPresentPaymentsOnboardingViewModel, rhs: CardPresentPaymentsOnboardingViewModel) -> Bool {
+        lhs.state == rhs.state &&
+        lhs.userIsAdministrator == rhs.userIsAdministrator &&
+        lhs.learnMoreURL == rhs.learnMoreURL
+    }
+}
```

Comment:
❓ A bit of a general question, since we cannot really fully conform the protocol and the closures, would still be worth to approximate it? For example:

```suggestion
extension CardPresentPaymentsOnboardingViewModel: Equatable {
    static func == (lhs: CardPresentPaymentsOnboardingViewModel, rhs: CardPresentPaymentsOnboardingViewModel) -> Bool {
        lhs.state == rhs.state &&
        lhs.userIsAdministrator == rhs.userIsAdministrator &&
        lhs.learnMoreURL == rhs.learnMoreURL
        lhs.useCase.state == rhs.useCase.state &&
        (lhs.showSupport != nil) == (rhs.showSupport != nil) &&
        (lhs.showURL != nil) == (rhs.showURL != nil)
    }
}
```
I don't have any specific use case in mind where performing an equal check between two onboarding view models could pass as false positive due to missing properties, it's more of "I'm curious" 😅

Reply:
This might be worth doing, and would slightly improve the conformance.

A bigger risk is sometimes that other properties could be added in future and missed from the equatable conformance, because the person adding them doesn't realise there's a custom function for it.

One thing I've added recently is a test function for counting properties – if you add a test asserting that the property count matches what it is right now, with a message saying "Please update the equatable conformance if this test fails", you give a chance to catch that issue in future.

[See `PointOfSaleCardPresentPaymentConnectingFailedUpdateAddressAlertViewModelTests` for an example.](https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/WooCommerceTests/POS/Card%20Present%20Payments/Connection%20Alerts/PointOfSaleCardPresentPaymentConnectingFailedUpdateAddressAlertViewModelTests.swift)

Reply:
I'm more used to comparing what represents the identity of a class - like the state in this case - without the closures and internal dependencies, but it looks safer to include the closure nullability checks. Great suggestion on adding a test case to fail when new properties are added. Updates in https://github.com/woocommerce/woocommerce-ios/pull/14224/commits/66dca3d68202c6773436835f7066559ad20fb799.